### ISSUE TITLE
[Arm32][Legacy Backend][JitStress=2] corefx test failure in System.Xml.XPath.XDocument.Tests.dll, more

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#13213]
Test failures in:
```
System.Xml.XPath.XDocument.Tests.dll
System.Xml.XPath.XmlDocument.Tests.dll
```

These exhibit failures for COMPlus_JitStress=2. The failures are all like:

```
      System.NullReferenceException : Object reference not set to an instance of an object.
      Stack Trace:
            at System.Xml.XmlTextReaderImpl.get_IsDefault()
            at System.Xml.XmlLoader.LoadAttributeNode()
            at System.Xml.XmlLoader.LoadNode(Boolean skipOverWhitespace)
            at System.Xml.XmlLoader.Load(XmlDocument doc, XmlReader reader, Boolean preserveWhitespace)
            at System.Xml.XmlDocument.Load(XmlReader reader)
            at System.Xml.XmlDocument.Load(Stream inStream)
            at XPathTests.CreateNavigatorFromXmlDocument.CreateNavigatorFromFile(String fileName)
            at XPathTests.Common.Utils.CreateNavigatorFromFile(String fileName)
            at XPathTests.Common.Utils.CreateNavigator(String xml, String startingNodePath, XmlNamespaceManager namespaceManager)
            at XPathTests.Common.Utils.XPathNodesetTest(String xml, String testExpression, XPathResult expected, XmlNamespaceManager namespaceManager, String startingNodePath)
            at XPathTests.FunctionalTests.CustomerScenariosTests.CustomerScenariosTest303()
```

A different number of tests, and a different set of individual tests, fails each time. Without JitStress, the tests pass.

If run under xunit with the `-maxthreads 1` argument, then the tests pass even with JitStress=2. This indicates possibly some kind of race condition where JitStress is violating some related invariant.

### How to build:
```
(coreclr)
build arm checked skiptests skipbuildpackages
(corefx)
build -Release -os:Windows_NT -buildArch:arm -- /p:CoreCLROverridePath=<coreclr_root>\bin\Product\Windows_NT.arm.Checked
build-tests -Release -os:Windows_NT -buildArch:arm -SkipTests -- /p:WithoutCategories=IgnoreForCI
```

### How to run:
```
(For any RunTests.cmd under <corefx_root>\bin)
       RunTests.cmd <corefx_root>\bin\testhost\netcoreapp-Windows_NT-Release-arm
```

### Example command line
```
Z:\gh\corefx\bin\Windows_NT.AnyCPU.Release\System.Xml.XPath.XmlDocument.Tests\netcoreapp>Z:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-arm\dotnet.exe xunit.console.netcore.exe System.Xml.XPath.XmlDocument.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonwindowstests  -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
```

**Comment by BruceForstall** [comment]
This appears to be a tailcall stress problem (either a problem with the stress mode, or a tailcall bug in the compiler). Setting:
```
set COMPlus_JitStressModeNamesNot=STRESS_TAILCALL
```
along with
```
set COMPlus_JitStress=1
```
allows the tests to pass.

**Comment by BruceForstall** [comment]
There is a similar failure with System.Xml.XPath.XDocument.Tests.dll only, with JitStress=1, that is also resolved by setting
```
set COMPlus_JitStressModeNamesNot=STRESS_TAILCALL
```

**Comment by BruceForstall** [comment]
Simply setting
```
set COMPlus_TailcallStress=1
```
causes System.Xml.XPath.XDocument.Tests.dll to fail, but not System.Xml.XPath.XmlDocument.Tests.dll.

**Comment by sandreenko** [comment]
I was not able to repro any of this on x86 with legacynonjit.

**Comment by rartemev** [comment]
Fixed in #14347
